landscaper/gardenlet/pkg/gardenlet_chart_values.go

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pkg
+
+import (
+	"github.com/gardener/gardener/pkg/gardenlet/controller/managedseed"
+)
+
+func (g Landscaper) computeGardenletChartValues(bootstrapKubeconfig []byte) (map[string]interface{}, error) {
+	// do not set the garden bootstrap secret to the chart when the actual bootstrap kubeconfig is nil
+	// otherwise this will lead to a chart render error
+	if bootstrapKubeconfig == nil && g.gardenletConfiguration.GardenClientConnection != nil {
+		g.gardenletConfiguration.GardenClientConnection.BootstrapKubeconfig = nil
+	}
+
+	// do not set any parent Gardenlet configuration as the landscaper
+	// does not have a parent Gardenlet
+	valuesHelper := managedseed.NewValuesHelper(nil, nil)
+	return valuesHelper.GetGardenletChartValues(g.imports.DeploymentConfiguration, g.gardenletConfiguration, string(bootstrapKubeconfig))
+}
```

Comment:
Wow, this is indeed much better! :-) I am only slightly unsure about the dependency to `pkg/gardenlet/controller/managedseed`, I think it would be a bit cleaner to put this helper in a util package somewhere, but this is a really minor nit.